### PR TITLE
dependabot: Fix docker/go paths

### PR DESCRIPTION
**Pull request by phlax**
Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -52,82 +52,82 @@ updates:
...
     interval: daily
 
 - package-ecosystem: "docker"
-  directory: "/examples/grpc-bridge"
+  directory: "/examples/mysql"
   schedule:
     interval: daily
 
 - package-ecosystem: "docker"
-  directory: "/examples/grpc-bridge/client"
```
Comment (alyssawilk) [inline .github/dependabot.yml:85]: am I failing at search or did we lose this one?

File: .github/dependabot.yml [hunk 2]
```diff
@@ -52,82 +52,82 @@ updates:
...
     interval: daily
 
 - package-ecosystem: "docker"
-  directory: "/examples/local_ratelimit"
+  directory: "/examples/shared/echo"
   schedule:
     interval: daily
 
 - package-ecosystem: "docker"
-  directory: "/examples/lua"
```
Comment (alyssawilk) [inline .github/dependabot.yml:105]: ditto?

**Review by alyssawilk — COMMENTED** [review]
If we accidentally lost any I'm going to say alphabetizing isn't worth it and we should just add any missing paths
/wait

File: .github/dependabot.yml (hunk 1 above)
Comment (phlax) [reply .github/dependabot.yml:85]: we moved the recipes into shared folders to limit the dependabot activity, improve docker layering and make maintenance easier see here https://github.com/envoyproxy/envoy/pull/26005/files

**Comment by phlax** [comment]
re this - i think the key thing is to get some CI in place so we dont break it etc (https://github.com/envoyproxy/envoy/issues/26163)

File: .github/dependabot.yml (hunk 2 above)
Comment (phlax) [reply .github/dependabot.yml:105]: i think `lua` was there for `echo` Dockerfile

**Review by alyssawilk — APPROVED** [review]
thanks for the comment!

**Pull request merged**